web_src/js/features/heatmap.js

```diff
@@ -17,7 +17,18 @@ export default function initHeatmap() {
       return {date: new Date(v), count: heatmap[v]};
     });
 
-    const View = createApp(ActivityHeatmap, {values});
+    const locale = {
+      months: new Array(12).fill().map((_, idx) => translateMonth(idx)),
+      days: new Array(7).fill().map((_, idx) => translateDay(idx)),
+      contributions: el.getAttribute('data-locale-contributions').toLocaleLowerCase(getCurrentLocale()),
+      no_contributions: el.getAttribute('data-locale-no_contributions'),
+      contributions_in_the_last_12_months: el.getAttribute('data-locale-contributions_in_the_last_12_months').toLocaleLowerCase(getCurrentLocale()),
+      on: el.getAttribute('data-locale-on').toLocaleLowerCase(getCurrentLocale()),
```

Comment:
Why lowercased?
Why not display it as is?

Reply:
That's how it was before.

"4 Contributions in the last 12 months" looks worse than "4 contributions in the last 12 months"

"3 Contributions On Oct 4, 2025" looks worse than "3 contributions on Oct 4, 2025" (tooltip)

Reply:
The problem is: I agree for the English locale, but that isn't necessarily the case for all locales…

Reply:
Isn't lowercase the best bet?

Do you prefer `toLowerCase` instead of `toLocaleLowerCase`?

Reply:
What we could of course do:
Since this is a relatively "non-issue", we can just test your approach (in production), and when people complain, we can revert it.
That makes probably the most sense without arguing about something so minor.

Reply:
I prefer to remove the letter case casting, leave the translation work fully to translators.

And use "%d contributions in the last 12 months" to make the text contains the number position to make the translation easier.

If there is still anything wrong in Heatmap, it should be fixed in Heatmap side, but not patched by Gitea  side.

Reply:
Agree, we should not translate single words of sentences because you can't just piece them together in all languages. Translations should be whole statements/sentences.

Reply:
Ok, I think I'll revert these back to how it was and keep this PR only on months and days